skfda/datasets/_samples_generators.py

```diff
@@ -1,26 +1,212 @@
+
+    See also:
+        :func:`make_gaussian_process`: Simpler function for generating
+        Gaussian processes.
+
+    Examples:
+        Example of the use of euler_maruyama for an SDE with constant
+        drift and diffusion and an initial condition which follows a
+        standard normal distribution.
+
+        >>> from scipy.stats import norm
+        >>> def example_drift(t: float, x: np.ndarray) -> np.ndarray:
+        ...     return 2
+        >>> def example_diffusion(t: float, x: np.ndarray) -> np.ndarray:
+        ...     return 0.5
+        >>> initial_condition = norm().rvs
+        >>>
+        >>> trajectories = euler_maruyama(
+        ...     initial_condition=initial_condition,
+        ...     n_samples=10,
+        ...     drift=example_drift,
+        ...     diffusion=example_diffusion,
+        ... )
+
+    """
+    random_state = validate_random_state(random_state)
+
+    if n_samples is None:
+        if callable(initial_condition):
+            raise ValueError(
+                "Invalid initial conditions. If a function is given, the "
+                "n_samples argument must be included.",
+            )
+
+        initial_values = np.atleast_1d(initial_condition)
+        n_samples = len(initial_values)
+    else:
+        if callable(initial_condition):
+            initial_values = initial_condition(
+                size=n_samples,
+                random_state=random_state,
+            )
+        else:
+            initial_condition = np.atleast_1d(initial_condition)
+            dim_codomain = len(initial_condition)
+            initial_values = (
+                initial_condition
+                * np.ones((n_samples, dim_codomain))
+            )
+
+    if initial_values.ndim == 1:
+        initial_values = initial_values[:, np.newaxis]
+    elif initial_values.ndim > 2:
+        raise ValueError(
+            "Invalid initial conditions. Each of the starting points "
+            "must be a flat array.",
+        )
+    (n_samples, dim_codomain) = initial_values.shape
+
+    if dim_noise is None:
+        dim_noise = dim_codomain
+
+    def default_drift(
+        t: float,
+        x: np.typing.NDArray[np.floating[Any]],
+    ) -> np.typing.NDArray[np.floating[Any]]:
+        return np.array(0)
+
+    def default_diffusion(
+        t: float,
+        x: np.typing.NDArray[np.floating[Any]],
+    ) -> np.typing.NDArray[np.floating[Any]]:
+        return np.eye(dim_codomain, dim_noise)
+
+    if drift is None:
+        drift = default_drift
+
+    if diffusion is None:
+        diffusion = default_diffusion
+
+    test_diffusion_dim = np.atleast_1d(diffusion(start, initial_values))
+
+    if test_diffusion_dim.ndim == 1:
+        if len(test_diffusion_dim) not in {dim_codomain, 1}:
+            raise ValueError(
+                f"The diffusion function returns the wrong dimensions."
+                f"The expected dimension is {dim_codomain} or 1.",
```

Comment:
Why 1?

Reply:
The value 1 references to the case were the diffusion coefficient is just a constant. In that case, test_diffusion_dim should be an array with one element and that would be a valid scenario.